src/libraries/System.Runtime/tests/System/Runtime/ControlledExecutionTests.cs

```diff
@@ -0,0 +1,199 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading;
+using Xunit;
+
+// Disable warnings for ControlledExecution.Run
+#pragma warning disable SYSLIB0046
+
+namespace System.Runtime.Tests
+{
+    public class ControlledExecutionTests
+    {
+        private bool _startedExecution, _caughtException, _finishedExecution;
+        private Exception _exception;
+        private CancellationTokenSource _cts;
+        private volatile int _counter;
+
+        // Tests cancellation on timeout. The ThreadAbortException must be mapped to OperationCanceledException.
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsWindows), nameof(PlatformDetection.IsNotNetFramework), nameof(PlatformDetection.IsNotMonoRuntime), nameof(PlatformDetection.IsNotNativeAot))]
```

Comment:
What prevents these tests from working on Unix?

Reply:
Missing `THROW_CONTROL_FOR_THREAD_FUNCTION` assembler helpers in `.S` files.

Reply:
We have issue opened on this https://github.com/dotnet/runtime/issues/72703.

Can we make most of these tests work on Unix using the patch suggested in https://github.com/dotnet/runtime/issues/12739#issuecomment-1192053346 ?

The tests that do not work even with this workaround should be disabled using `[ActiveIssue("https://github.com/dotnet/runtime/issues/72703", TestPlatforms.AnyUnix)]`.

Reply:
> Can we make most of these tests work on Unix using the patch suggested in [#12739 (comment)](https://github.com/dotnet/runtime/issues/12739#issuecomment-1192053346) ?

Yes, 5 out of 6 tests are passing now. Thanks!

Reply:
The tests always pass on my Ubuntu; however, they failed in 3 CI legs:

> libc++abi: terminating with uncaught exception of type PAL_SEHException

and

> terminate called after throwing an instance of 'PAL_SEHException'
```
0:000> kc
 # Call Site
00 libc_2_28!gsignal
01 libc_2_28!abort
02 libstdc___so_6_0!std::__once_callable
03 libstdc___so_6_0!std::rethrow_exception
04 libstdc___so_6_0!std::terminate
05 libstdc___so_6_0!_cxa_rethrow
06 libcoreclr!RaiseTheExceptionInternalOnly
07 libcoreclr!Thread::HandleThreadAbort
08 libcoreclr!HandleSuspensionForInterruptedThread
09 libcoreclr!inject_activation_handler
0a libpthread_2_28!funlockfile
0b 0x0
```
I guess the `ThreadAbortException` is escaping somehow.  There is only one crash dump and I could not extract useful information from it besides the call stack above.  I think we may disable this for Unix for now and follow up later.

Reply:
I figured out the bottom stack frame here is one of my tests and the next frame (incorrectly labeled `funlockfile`), is a `syscall(15)` call, which is `sigreturn`. I guess we corrupt the call stack, which makes the exception go unhandled.